src/manifests/input_manifest.py

```diff
@@ -49,6 +54,9 @@ class InputManifest(ComponentManifest):
                 "name": {"required": True, "type": "string"},
                 "version": {"required": True, "type": "string"},
                 "patches": {"type": "list", "schema": {"type": "string"}},
+                "platform": {"type": "string"},
+                "architecture": {"type": "string"},
+                "snapshot": {"type": "boolean"},
             },
         },
         "ci": {
```

Comment:
What about the notion of the opensearch-build repositories state?  If there are changes to this repo the build might not be stable, if we had BWC tests I'd have more confidence, but I'm not sure we want to go to that length

Reply:
Note; if we wanted to manage this less broadly, what about including the SHA of the source manifest file?

Reply:
There's 100% chance that this repo will have changed a month after, say, a 1.0 release. But that's just tooling and shouldn't need to rebuild 1.0.

Reply:
We could include the SHA of the original manifest file, but how and why would we use that?

Reply:
You've convinced me, I don't think we need to be overzealous in the reproducibility space, especially since jenkins isn't going to have a stable ref for a considerable time.